driver/sccb-ng.c

```diff
@@ -368,7 +258,12 @@ int SCCB_Write_Addr16_Val16(uint8_t slv_addr, uint16_t reg, uint16_t data)
     tx_buffer[2] = data >> 8;
     tx_buffer[3] = data & 0x00ff;
 
-    esp_err_t ret = i2c_master_transmit(dev_handle, tx_buffer, 4, TIMEOUT_MS);
+    esp_err_t ret;
+    i2c_master_dev_handle_t dev_handle;
+
+    register_device(slv_addr, &dev_handle);
+    ret = i2c_master_transmit(dev_handle, tx_buffer, 4, TIMEOUT_MS);
```

Comment:
Does the result of `register_device` need to be checked?

if we cared about the repeated code, could be factored out to a function such as `i2c_master_transmit_unregistered`.

Reply:
The current error handling for `SCCB_Write()`, etc. which call `register_device()` isn't very good either. I just copied that style. So this is no better and no worse than the rest of the code in the driver.

Yes, I could do some refactoring. I was trying to make minimal changes to the original driver.